rpc/connection.go

```diff
@@ -627,8 +627,13 @@ func (c *Connection) connect(ctx context.Context) error {
 }
 
 // DoCommand executes the specific rpc command wrapped in rpcFunc.
-func (c *Connection) DoCommand(ctx context.Context, name string,
+func (c *Connection) DoCommand(ctx context.Context, name string, timeout time.Duration,
 	rpcFunc func(GenericClient) error) error {
+	if timeout > 0 {
+		var timeoutCancel context.CancelFunc
+		ctx, timeoutCancel = context.WithTimeout(ctx, timeout)
+		defer timeoutCancel()
+	}
 	for {
 		if c.initialReconnectBackoffWindow != nil && isWithFireNow(ctx) {
```

Comment:
check for ctx.Done here? or does some function within this do that?

Reply:
Sure

Reply:
Actually don't think we need to, the relevant places check it

Reply:
great